cli/build.rs

```diff
@@ -22,20 +22,25 @@ const JJ_OP_HEADS_PATH: &str = "../.jj/repo/op_heads/heads";
 fn main() {
     let version = std::env::var("CARGO_PKG_VERSION").unwrap();
 
-    if Path::new(GIT_HEAD_PATH).exists() {
-        // In colocated repo, .git/HEAD should reflect the working-copy parent.
-        println!("cargo:rerun-if-changed={GIT_HEAD_PATH}");
-    } else if Path::new(JJ_OP_HEADS_PATH).exists() {
-        // op_heads changes when working-copy files are mutated, which is way more
-        // frequent than .git/HEAD.
-        println!("cargo:rerun-if-changed={JJ_OP_HEADS_PATH}");
-    }
     println!("cargo:rerun-if-env-changed=NIX_JJ_GIT_HASH");
+    let git_hash = get_git_has_from_nix()
+        .or_else(|| {
+            if Path::new(JJ_OP_HEADS_PATH).exists() {
+                // op_heads changes when working-copy files are mutated, which is way more
+                // frequent than .git/HEAD.
+                println!("cargo:rerun-if-changed={JJ_OP_HEADS_PATH}");
+            }
+            get_git_hash_from_jj()
```

Comment:
Doesn't this require more frequent rebuilds? In the original code, we prioritize `.git/HEAD` because it's more stable than op heads.

Reply:
Ah I misunderstood the priority order (since `get_git_hash` checks `jj` before checking `git`). Since `.git/HEAD` changes more infrequently these should be swapped so the `git` one comes first

Reply:
Ah I understand now: when telling cargo to rebuild we should prioritize `GIT_HEAD_PATH` if present, and only if its missing, then rebuild when `JJ_OP_HEADS_PATH` changes. But when it comes to pulling the actual hash, prefer using `jj` if installed, otherwise fall back to `git`. I've updated the code to reflect this!